src/utils/airtable/interface.ts

```diff
@@ -0,0 +1,183 @@
+export type TableValues = string | boolean | number | Array<unknown> | Airtable.Attachment;
+export type TableRecord =
+  | UserRecord
+  | CustomerRecord
+  | SiteRecord
+  | InvoiceRecord
+  | PaymentRecord
+  | CustomerUpdateRecord
+  | MeterReadingRecord
+  | TariffPlanRecord
+  | InventoryRecord
+  | InventoryUpdateRecord
+  | IncidentRecord
+  | IncidentUpdateRecord
+  | MaintenanceRecord
+  | FinancialReportRecord;
+
+export type Row = {
+  get: (key: string) => TableValues;
+  getId: () => string;
+};
+
+interface Record {
+  rid: string;
+}
+
+export interface UserRecord extends Record {
+  အမည်: string;
+  name: string;
+  username: string;
+  email: string;
+  password: string;
+  photo?: Airtable.Attachment[];
+  incidentIds: string[];
```

Comment:
Should some of these properties in these interfaces be optional (ex: Users aren't required to have an email)? It might make sense to establish the optional-ness later, though. It seems like a lot of invariants to enforce right now before we understand how our data really works.

Reply:
yeah, i didn't want to define optional-ness for certain properties because i wasn't 100% sure what would/wouldn't be necessary given how the data works but definitely something to adjust later on!